rest_framework/serializers.py

```diff
@@ -1002,8 +1003,16 @@ def get_fields(self):
                     field_cls = self._get_nested_class(depth, relation_info)
                     kwargs = get_nested_relation_kwargs(relation_info)
                 else:
-                    field_cls = self._related_class
                     kwargs = get_relation_kwargs(field_name, relation_info)
+                    to_field = kwargs.pop('to_field', None)
+                    # it seems that some tests/django initializers are setting
+                    # to_field where it is totally unnecessary
+                    if to_field and to_field != 'id':
```

Comment:
I think it also has to do with what gets into that to_field and makes it to that point. I would remove any 'id' entries in compat/get_to_field making this check here unnecessary. Maybe not the best solution but the to_field irritates me for now. I think for plain ForeignKeys it is safe but for anything else... i dont know.

I added some links to the core issue of that 2 posts above. Thats what i meant.